index.js

```diff
@@ -1127,3 +1137,20 @@ function setHeaders (res, headers) {
     res.setHeader(key, headers[key])
   }
 }
+
+/**
+ * Check if the given file incusive the path exits with the check on case sensitivity
+ * @param filepath the file to be checked
+ * @return {boolean} true when a file with the same case exists false if not
+ */
+function fileExistsWithCaseSync (filepath) {
+  var dir = path.dirname(filepath)
+  if (dir === path.dirname(dir)) {
+    return true
+  }
+  var filenames = fs.readdirSync(dir)
```

Comment:
We should really do this using an async method; sync methods will block the event loop and processing of all I/O in the process while the read occurs, and it could be against a network share, for example.